src/ptunnel.c

```diff
@@ -668,7 +668,7 @@ void* pt_proxy(void *args) {
 				             cur->dst_ip, cur->dst_port, kProto_ack | cur->type_flag,
 				             &cur->dest_addr, cur->next_remote_seq, &cur->send_first_ack, &cur->ping_seq, cur->window_size);
 				cur->xfer.icmp_ack_out++;
-				if (cur->send_ring[idx].pkt_len > sizeof(icmp_echo_packet_t) && cur->send_ring[idx].pkt->type == 8) {
```

Comment:
Shouldn't pkt_len > sizeof(icmp_echo_packet_t) have the same effect?

Reply:
It would have the same effect except pkt_len is not cleared when slots in send_ring are reused.

The alternate solution is to set pkt_len to zero.